core/data/path.go

```diff
@@ -41,6 +42,15 @@ func GetAttrPath(inAttrName string) (attrName string, attrPath string, pathType
 				attrPath = inAttrName[idx+2:]
 			}
 		}
+	} else if inAttrName[0] == '$' && inAttrName[1] == '{' {
```

Comment:
We should use strings.HasPrefix() function here. That will protect against inAttrName smaller than 2 in len().

In fact, if you check for prefixes '${property.' || '${env.' you remove quite a bit of code there, and will not affect any other part of the if statement.

} else if strings.HasPrefix('${property.') || strings.HasPrefix('${env.'){